src/util.cpp

```diff
@@ -85,14 +85,30 @@ class CInit
             ppmutexOpenSSL[i] = new CCriticalSection();
         CRYPTO_set_locking_callback(locking_callback);
 
-#ifdef WIN32
-        // Seed OpenSSL PRNG with current contents of the screen
-        RAND_screen();
-#endif
+        // Check whether OpenSSL random number generator is properly seeded. If not, attempt to seed using RAND_poll().
+        // Note that in versions of OpenSSL in the depends in Gridcoin (currently 1.1.1+), and modern Unix distros (using
+        // openSSL 1.1.1+ or modern Windows operating systems with the equivalent of /dev/urandom, the random number
+        // generator is automatically seeded on init, and periodically reseeded from trusted OS random sources. It is
+        // not necessary to manually reseed the RNG. Here we implement a check via RAND_status() to ensure the RNG
+        // is properly seeded. If not, we try 10 times (probably excessive) to seed the RNG via openSSL's entropy sources,
+        // breaking as soon as the return from RAND_poll() becomes 1 (successfully seeded). A 100 ms sleep is inserted
+        // between each try to ensure we give time for a short term unavailability of the OS entropy source to recover.
+        // If this falls through, we abort the application as we cannot have a non-functioning RNG.
+        bool seed_successful = RAND_status();
+        if (!seed_successful) {
+            for (unsigned int i = 0; i < 10; ++i)
+            {
+                seed_successful = RAND_poll();
 
-        // Seed OpenSSL PRNG with performance counter
-        RandAddSeed();
+                if (seed_successful) break;
+
+                MilliSleep(100);
+            }
+
+            if (!seed_successful) std::abort();
```

Comment:
This is an unlikely condition to reach but I would prefer if the user could be alerted that the RNG is not able to be seeded, rather than just aborting.